play-json/jvm/src/main/scala/play/api/libs/json/BigDecimalParser.scala

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (C) 2009-2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package play.api.libs.json
+
+import scala.util.control.NonFatal
+
+object BigDecimalParser {
+
+  def parse(input: String, settings: JsonParserSettings): Either[Error, java.math.BigDecimal] = {
+
+    // There is a limit of how large the numbers can be since parsing extremely
+    // large numbers (think thousand of digits) and operating on the parsed values
+    // can potentially cause a DDoS.
+    if (input.length > settings.bigDecimalParseSettings.digitsLimit) {
+      Left(Error.DigitLimitError())
```

Comment:
Why introducing a new `Error` type instead of "just" (KISS) having the error key (`String`) ?

Reply:
The errors are transformed into different things in json parser (exceptions) and reader JsResult.

Reply:
The jackson code can pattern match on the string code and the `JsResult` can just propagate the key.

Reply:
1) I will create `val`s then for those string values so that I can reference them in multiple places. So there will be some code listing all possible errors.
2) `ScaleLimitError(scale: Int)` carry error context which is handy.